tests/LowestCommonAncestorTest.php

```diff
@@ -0,0 +1,143 @@
+        $this->assertEquals('node_2', $test_node1->getId());
+        $test_node2 = $lca->find($vertices['node_3'], $vertices['node_4']);
+        $this->assertEquals('node_2', $test_node2->getId());
+        $test_node3 = $lca->find($vertices['node_6'], $vertices['node_4']);
+        $this->assertEquals('node_4', $test_node3->getId());
+        $test_node4 = $lca->find($vertices['node_6'], $vertices['node_7']);
+        $this->assertEquals('node_3', $test_node4->getId());
+        $test_node5 = $lca->find($vertices['node_6'], $vertices['node_5']);
+        $this->assertEquals('node_2', $test_node5->getId());
+
+    }
+
+    //Graphical representation for the Most Complex Graph is in pictures folder.
+    public function testmostComplexGraph()
+    {
+        $graph = new Graph();
+        //Creating new Graph with 9 nodes.
+        $vertices = $this->generateVertices($graph, 10);
+        //Creating edges starting from root(node_1) to node_9
+        $vertices['node_1']->createEdgeTo($vertices['node_2']);
+        $vertices['node_1']->createEdgeTo($vertices['node_3']);
+        $vertices['node_1']->createEdgeTo($vertices['node_4']);
+        $vertices['node_2']->createEdgeTo($vertices['node_5']);
+        $vertices['node_2']->createEdgeTo($vertices['node_7']);
+        $vertices['node_3']->createEdgeTo($vertices['node_6']);
+        $vertices['node_3']->createEdgeTo($vertices['node_7']);
+        $vertices['node_3']->createEdgeTo($vertices['node_5']);
+        $vertices['node_4']->createEdgeTo($vertices['node_6']);
+        $vertices['node_4']->createEdgeTo($vertices['node_8']);
+        $vertices['node_5']->createEdgeTo($vertices['node_7']);
+        $vertices['node_5']->createEdgeTo($vertices['node_8']);
+        $vertices['node_6']->createEdgeTo($vertices['node_7']);
+        $vertices['node_6']->createEdgeTo($vertices['node_8']);
+
+        $lca = new LowestCommonAncestor($graph);
+
+        $test_node1 = $lca->find($vertices['node_5'], $vertices['node_7']);
+        $this->assertEquals('node_5', $test_node1->getId());
+        $test_node2 = $lca->find($vertices['node_7'], $vertices['node_8']);
+        $this->assertEquals('node_2', $test_node2->getId());
+        //Same node passed as remote and local
+        $test_node3 = $lca->find($vertices['node_7'], $vertices['node_7']);
+        $this->assertEquals('node_7', $test_node3->getId());
+        $test_node4 = $lca->find($vertices['node_2'], $vertices['node_8']);
+        $this->assertEquals('node_2', $test_node4->getId());
+        $test_node5 = $lca->find($vertices['node_8'], $vertices['node_2']);
+        $this->assertEquals('node_1', $test_node5->getId());
+        //For the node with no parent: No common parents found.
+        try {
+            $test_node6 = $lca->find($vertices['node_8'], $vertices['node_10']);
+            $this->fail('Exception was not thrown.');
+        }
+        catch (UnderflowException $e){
+            $this->pass('Exception was thrown correctly');
```

Comment:
call to undefined method pass(). I find it strange, it was working just fine. Any tips?